common-content/en/module/tools/single-use-data-analysis/index.md

```diff
@@ -0,0 +1,83 @@
++++
+title = "Single-use data analysis programs"
+headless = true
+time = 60
+facilitation = false
+emoji= "💻"
+[objectives]
+    1="Write a program to extract information from a JSON file"
+    2="Identify the trade-offs between using existing shell tools and writing custom programs"
+    3="Choose whether to use existing tools or write a custom program to solve a particular problem"
++++
+
+We've seen two different ways of analysing some input to produce an output.
+
+Sometimes we can use, or combine, existing tools to get answers. For instance, we can count the words in a file with `wc`.
+
+Sometimes we can write custom tools when existing tools don't quite do what we want. For instance, we wrote a program to count specific words.
+
+When we want to answer some question, sometimes it's useful to write a program that we may only use one time. Or we may re-use in the future.
+
+It's not always obvious whether it's easier to try to use tools that already exist, or to write our own.
```

Comment:
Crying out for a concrete analogy here. Can you find an example in ordinary life like this? Sometimes it's easier to just make ourselves a sandwich than to go to a supermarket in a foreign country. Or something.

Reply:
I tried, with funnels...!